packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -50,6 +50,25 @@ async function getClimateDataset(datasetId: string): Promise<Dataset | null> {
 	return response?.data ?? null;
 }
 
+async function getClimateSubsetId(
+	esgfId: string,
+	parentDatasetId: string,
+	envelope: string,
+	timestamps?: string,
+	thinFactor?: string
+): Promise<string | null> {
```

Comment:
This functional interface design is unclear, if you have consecutive optionals they become difficult to reason with. e.g. if I want to specify `thinFactor` but not `timestamps` then timestamps is no longer optional.

If you have a bunch of things that may or may not be specified, place them into an options object:
```
function abc( a: string, b: string, c: number, options: { .... })
```